### PR TITLE
Wordlist config generator

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do?

Adds creds from wordlists to config

### CODE REVIEW

File: envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import copy
+import json
+import os
+from pathlib import Path
+
+import nanoid as nanoid
+
+parser = argparse.ArgumentParser(
+    prog="Credential Config Generator",
+    description="Small script to add usernames and " "passwords to Infection Monkey configuration",
```
Comment (shreyamalviya) [inline envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py:11]:
```suggestion
    description="Small script to add usernames and passwords to Infection Monkey configuration",
```

File: envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+)
+
+parser.add_argument(
+    "-u",
+    "--userlist-path",
+    type=str,
+    required=True,
+    help="Path to userlist file. Userfile should contain one username per line.",
+)
```
Comment (shreyamalviya) [inline envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py:20]: This is kind of confusing.
```suggestion
parser.add_argument(
    "-u",
    "--usernamelist-path",
    type=str,
    required=True,
    help="Path to file with list of usernames. File should contain one username per line.",
)
```

File: envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+parser.add_argument(
+    "-p",
+    "--passwordlist-path",
+    type=str,
+    required=True,
+    help="Path to passwordlist file. " "Passwordfile should contain one password per line.",
+)
```
Comment (shreyamalviya) [inline envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py:27]: Same as above
```suggestion
parser.add_argument(
    "-p",
    "--passwordlist-path",
    type=str,
    required=True,
    help="Path to file with list of passwords. File should contain one password per line.",
)
```

File: envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+parser.add_argument(
+    "-c",
+    "--config-path",
+    type=str,
+    help="Path to Infection Monkey configuration file."
+    "Must be unencrypted. If not provided, will use default config from a file.",
```
Comment (shreyamalviya) [inline envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py:33]: Where the default config is stored isn't relevant to the user.
```suggestion
    "Must be unencrypted. If not provided, default config will be used.",
```

**Review by shreyamalviya — CHANGES_REQUESTED** [review]

File: envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    # NanoId with 21 characters
+    "id": "",
+    "identity": "",
+    "password": "",
+    "lm": "",
+    "ntlm": "",
+    "ssh_public_key": "",
+    "ssh_private_key": "",
+    "isNew": True,
+}
```
Comment (shreyamalviya) [inline envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py:53]: Isn't the config generated here going to be sent to the propagation credentials endpoint directly?

Comment (VakarisZ) [reply envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py:53]: Yes, during configuration import credentials are sent to the endpoint

Comment (shreyamalviya) [reply envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py:55]: Then the format of this should be `{"identity": "", "secret": {...}}`. That's what the endpoint expects. The format here right now is how the frontend parses it.

**Review by mssalvatore — CHANGES_REQUESTED** [review]
Why are we adding this to our code base?

**Comment by VakarisZ** [comment]
It's not worth to add to the main codebase, but it's likely to work without any changes for many years to come and is a script that we could drop for users in case someone wants to use wordlists instead of our UI. Opening a PR makes sure we can find it if we need to in the future

**Pull request closed without merging**